src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -46,26 +46,6 @@ ref Unsafe.Add(ref strA.GetRawStringData(), (nint)(uint)indexA /* force zero-ext
                 ref Unsafe.Add(ref strB.GetRawStringData(), (nint)(uint)indexB /* force zero-extension */), countB);
         }
 
-        internal static bool EqualsOrdinalIgnoreCase(string? strA, string? strB)
```

Comment:
The issue here was that the JIT recognizes `string.Equals(..., StringComparison)` as an intrinsic but not this helper? Are there any situations now where, when the intrinsic doesn't kick in, going through string.Equals will be slower? Presumably this helper existed to avoid needing to validate and map the StringComparison?

Reply:
> Are there any situations now where, when the intrinsic doesn't kick in, going through string.Equals will be slower?

Ideally, it should always be inlined when the StringComparer is a constant argument, then JIT sees that it's fed into a switch opcode and gives it a high boost for inlining